project/.github/workflows/ci.yml

```diff
@@ -3,10 +3,10 @@ name: ci
 on:
   push:
     branches:
-      - master
+      - main
   pull_request:
     branches:
-      - master
+      - main
```

Comment:
Not totally sure about this. True, the default branch on new repositories on GitHub is now `main`, but locally many users have configured git to stick to `master` (I believe I did since my most recent repositories have a `master` branch). Ideally we would detect the configured default branch name and inject it into the context variables.
Or we could consider adding a question to the template so the user can specify their default branch.

Reply:
Question is probably best option. I will revert this change then so that you can eventually add a question as time permits :)

Reply:
@pawamoy 
So this should be all set now with the minimal stuff for python3.10 and precommit updates.

When doing the question for the default branch i think default should be whatever gitlab/github has which is main so when people complacently accept it just works but that as you mentioned will still allow others to set master that want that.

File that are affected by that with hardcoded values currently for reference. Note two of these were not using jinja2 before not sure if that is helpful to know or not but figured id mention it. ;)
```
project/README.md.jinja
project/.gitlab-ci.yml
project/.github/workflows/ci.yml
```

Reply:
Great, thanks! No default will *always* work, but yeah, we can settle for `main` as default for the future branch question.